File: B_Pile_Shuffling.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"
#define bijli() (ios_base:: sync_with_stdio(false),cin.tie(NULL), cout.tie(NULL));
#define ll long long int
#define all(x) x.begin(),x.end()

// void solution(){
//     ll n; cin >> n;
//     ll a, b, c, d;
//     ll ans = 0;
//     for(int i = 0; i < n; i++){
//         cin >> a >> b >> c >> d;
//         if(a>c){
//             ans+=(a-c);
//             a-=(a-c);
//         }
//         if(b>d && a == 0)
//             ans+=(b-d);
//         else if(b>d && a!=0)
//             ans+=(b-d+a);
//     }
//     cout << ans << endl;
    
// }

void solution(){
    ll n; cin >> n;
    ll a, b, c, d;
    ll ans = 0;
    
    vector <vector<ll>> l(n);
    for(int i = 0; i < n; i++){
        cin >> a >> b >> c >> d;
        l[i] = {a, b, c, d};
    }
    for(auto &v : l){
        
        if(v[0]>v[2]){
            ans+=(v[0]-v[2]);
            v[0]-=(v[0]-v[2]);
        }
        if(v[1]>v[3] && v[0] == 0)
            ans+=(v[1]-v[3]);
        else if(v[1]>v[3] && v[0]!=0)
            ans+=(v[1]-v[3]+v[0]);
    }

    cout << ans << endl;
    
    
    
}


int main(){
    bijli();
    int tc=1;
    cin >> tc;
    while(tc--){
        solution();
    }
    return 0;
}

File: B_Pile_Shuffling.py
# def solution():
#     n = int(input())
#     ls = []
#     for i in range (0,n):
#         a, b, c, d = map(int, input().split())
#         ls.append([a, b, c, d])

#     ans = 0
#     for l in ls:

#         if l[0]>l[2]:
#             ans+=(l[0]-l[2])
#             l[0] -= (l[0]-l[2])
#         if l[1] > l[3] and l[0] == 0:
#             ans+=(l[1]-l[3])
#         elif l[1] > l[3] and l[0] != 0:
#             ans+=(l[1]-l[3]+l[0])
#     print(ans)
def solution():
    n = int(input())
    ls = []
    ans = 0
    for i in range (0,n):
        a, b, c, d = map(int, input().split())
        if a > c:
            ans+=(a-c)
            a-=(a-c)
        if b > d and a == 0:
            ans+=(b-d)
        elif b > d and a != 0:
            ans+=(b-d+a)
    print(ans)


tc = 1
tc = int(input())
for _ in range(tc):
    solution()

